docs/source/data/subscriptions.mdx

```diff
@@ -42,102 +40,94 @@ subscription PostFeed {
-      const wsServer = new WebSocketServer({
-        // This is the `httpServer` we created in a previous step.
-        server: httpServer,
-        // Pass a different path here if your ApolloServer serves at
-        // a different path.
-        path: '/graphql',
-      });
-
-      // Hand in the schema we just created and have the
-      // WebSocketServer start listening.
-      const serverCleanup = useServer({ schema }, wsServer);
-    ```
-
-6. Add [plugins](../integrations/plugins) to your `ApolloServer` constructor to shutdown both the HTTP server and the `WebSocketServer`:
-
-    ```ts title="index.ts"
-      const server = new ApolloServer({
-        schema,
-        csrfPrevention: true,
-        cache: "bounded",
-        plugins: [
-          // Proper shutdown for the HTTP server.
-          ApolloServerPluginDrainHttpServer({ httpServer }),
-
-          // Proper shutdown for the WebSocket server.
-          {
-            async serverWillStart() {
-              return {
-                async drainServer() {
-                  await serverCleanup.dispose();
-                },
-              };
-            },
-          },
-          ApolloServerPluginLandingPageLocalDefault({ embed: true }),
-        ],
-      });
-    ```
+   ```ts title="index.ts"
+   // Creating the WebSocket server
+   const wsServer = new WebSocketServer({
+     // This is the `httpServer` we created in a previous step.
+     server: httpServer,
+     // Pass a different path here if your ApolloServer serves at
+     // a different path.
+     path: '/graphql',
+   });
+
+   // Hand in the schema we just created and have the
+   // WebSocketServer start listening.
+   const serverCleanup = useServer({ schema }, wsServer);
+   ```
+
+6. Add [plugins](/apollo-server/integrations/plugins) to your `ApolloServer` constructor to shutdown both the HTTP server and the `WebSocketServer`:
+
+   ```ts title="index.ts"
+   const server = new ApolloServer({
+     schema,
+     plugins: [
+       // Proper shutdown for the HTTP server.
+       ApolloServerPluginDrainHttpServer({ httpServer }),
+
+       // Proper shutdown for the WebSocket server.
+       {
+         async serverWillStart() {
+           return {
+             async drainServer() {
+               await serverCleanup.dispose();
+             },
+           };
+         },
+       },
+     ],
+   });
+   ```
 
 7. Finally, ensure you are `listen`ing to your `httpServer`.
 
-    Most Express applications call `app.listen(...)`, but for your setup  change this to `httpServer.listen(...)` using the same arguments. This way, the server starts listening on the HTTP and WebSocket transports simultaneously.
+   Most Express applications call `app.listen(...)`, but for your setup change this to `httpServer.listen(...)` using the same arguments. This way, the server starts listening on the HTTP and WebSocket transports simultaneously.
 
 A completed example of setting up subscriptions is shown below:
```

Comment:
Do you think it's actually useful that we walk through the code bit by bit and then show it all again here? Maybe just showing it once (fully complete) and not walking through (with extra comments added) would be better?

Reply:
Hmmm, it does feel like a bit much, but on the other hand, I could see this being confusing to implement, and at least this way, we **know** folks can't miss a step. I could go either way on this, honestly.